src/Microsoft.Diagnostics.Runtime/src/DataTargets/ClrInfo.cs

```diff
@@ -152,52 +52,7 @@ public override string ToString()
             return Version.ToString();
         }
 
-        internal ClrInfo(DataTargetImpl dt, ClrFlavor flavor, ModuleInfo module, DacInfo dacInfo, string dacLocation)
-        {
-            Debug.Assert(dacInfo != null);
-
-            Flavor = flavor;
-            DacInfo = dacInfo;
-            ModuleInfo = module;
-            module.IsRuntime = true;
-            _dataTarget = dt;
-            LocalMatchingDac = dacLocation;
-        }
-
-        internal ClrInfo()
-        {
-        }
-
-        private readonly DataTargetImpl _dataTarget;
-
-        /// <summary>
-        /// IComparable.  Sorts the object by version.
-        /// </summary>
-        /// <param name="obj">The object to compare to.</param>
-        /// <returns>-1 if less, 0 if equal, 1 if greater.</returns>
-        public int CompareTo(object obj)
-        {
-            if (obj == null)
-                return 1;
-
-            if (!(obj is ClrInfo))
-                throw new InvalidOperationException("Object not ClrInfo.");
-
-            ClrFlavor flv = ((ClrInfo)obj).Flavor;
-            if (flv != Flavor)
-                return flv.CompareTo(Flavor); // Intentionally reversed.
-
-            VersionInfo rhs = ((ClrInfo)obj).Version;
-            if (Version.Major != rhs.Major)
-                return Version.Major.CompareTo(rhs.Major);
-
-            if (Version.Minor != rhs.Minor)
-                return Version.Minor.CompareTo(rhs.Minor);
-
-            if (Version.Revision != rhs.Revision)
-                return Version.Revision.CompareTo(rhs.Revision);
-
-            return Version.Patch.CompareTo(rhs.Patch);
-        }
+        //for backward compatibility only
+        internal DataTarget DataTarget { get; }
```

Comment:
Marking this public isn't a bad idea actually.  I like having "back-references" up the object chain to the object's logical parent.